operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ConfigurationService.java

```diff
@@ -261,6 +253,11 @@ static ConfigurationService newOverriddenConfigurationService(
     return baseConfiguration;
   }
 
+  static ConfigurationService newOverriddenConfigurationService(
```

Comment:
This feels little smelly, since in an abstraction we creating an overrode of specific implementation, is this really needed?

Reply:
Not really, just noticing that the overriding method is almost always called passing it a new BaseConfigurationService instance. Also, I think it makes it easier for users because they might not know which instance to use to override the defaults. Open to cleaning this up further.